tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -764,7 +764,7 @@ def generate_gitlab_full_configuration(
 
     # Override some variables with a dedicated context
     if context:
-        full_configuration['variables'] = full_configuration.get('variables', {}).update(context)
+        full_configuration.get('variables', {}).update(context)
```

Comment:
The fix on the `linter.gitlab-ci` is here. Otherwise variables are set to `None`